src/utils/utils.rs

```diff
@@ -587,6 +587,22 @@ where
     })
 }
 
+pub fn delete_dir_contents(dir_path: &Path) {
+    let downloads_dir = fs::read_dir(&dir_path);
```

Comment:
It's not necessarily a downloads_dir at this point, so perhaps just call this `dir_contents` ?